### PR TITLE
External/bugfixes

### PR DESCRIPTION
**Pull request by iulianpascalau**
- added a series of linter fixes and small bugfixes

### CODE REVIEW

File: p2p/libp2p/directSender_test.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ import (
 	"github.com/libp2p/go-libp2p-core/peer"
 	"github.com/libp2p/go-libp2p-core/protocol"
 	pubsub "github.com/libp2p/go-libp2p-pubsub"
-	"github.com/libp2p/go-libp2p-pubsub/pb"
+	pubsub_pb "github.com/libp2p/go-libp2p-pubsub/pb"
```
Comment (AdoAdoAdo) [inline p2p/libp2p/directSender_test.go:24]: remove snakecase?

Comment (sasurobert) [reply p2p/libp2p/directSender_test.go:24]: +1

**Review by sasurobert — COMMENTED** [review]
good job!

File: p2p/libp2p/directSender_test.go (hunk 1 above)
Comment (iulianpascalau) [reply p2p/libp2p/directSender_test.go]: done in #2673

**Review by sasurobert — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Pull request merged**